### DISCUSSION TITLE
Combined energy data including microinverter on smart load input

### DISCUSSION THREAD

**Discussion by PMPoulsen** [davidrapan/ha-solarman#149] [Ideas]
Hi,

Is there a way to combine the accumulated PV power production for the Deye hybrid inverter when using the gen port for microinverters?
The production is somehow not included in the total PV production. The current power is reported as the smartload L1 power parameter but is not included in any accumulated data register.

My main reason is that I would like to record the total amount of PV power produced accumulative, and enter this in to the HA Energy section for the PV section. Currently I am using the total production parameter, but this do not include the microinverter production.

any ideas?

**Comment by davidrapan** [comment]
My opinion is that the microinverter should produce that info on it's own. Otherwise you will need to use custom integration sensor in the HA which can measure power over data from SmartLoad/Generator I/O.

**Reply by PMPoulsen** [reply]
I was just hoping that there may have been a "secret" register now that the inverter allow this function. The "micro" inverter (old string inverter) I use do not have this capability, so I will have to add a CT measurement device instead then.

**Reply by davidrapan** [reply]
If it's not included in the Total Production there is no other way.

You can use https://www.home-assistant.io/integrations/integration/ + https://www.home-assistant.io/integrations/utility_meter/ and "SmartLoad Power" as a source of the data. And that will functionally act as mentioned "CT measurement device". 😉

**Reply by PMPoulsen** [reply]
Will try this suggestion first, thanks
